app/models/queue_item.rb

```diff
@@ -23,18 +31,12 @@ def category_name
 
   private
 
-  def set_position
-    queue = self.user ? self.user.queue_items : [] # FIXME: just to pass the uniqueness shoulda matcher tests!!!
-    max_position = queue.empty? ? 0 : queue.count
-    self.position = max_position + 1
+  def review
+    @review ||= Review.where(creator: user, video: video).first
   end
 
-  def reassign_positions
-    remaining_queue = QueueItem.all.select do |queue_item|
-      queue_item.user == self.user and queue_item.position > self.position
-    end
-    remaining_queue.each do |queue_item|
-      queue_item.update(position: queue_item.position - 1)
-    end
+  def set_position
+    queue = QueueItem.where(user: self.user)
```

Comment:
Yes. done. But why doesn't the it work when i take out the self. on the very next line where i am trying to set the queue_item's position to default?